src/main/java/org/wildfly/security/credential/store/impl/KeystorePasswordStore.java

```diff
@@ -0,0 +1,573 @@
+    /**
+     * Value of this attribute denotes if this store can preform modification of its storage.
+     */
+    public static final String MODIFIABLE = "store.modifiable";
+    /**
+     * Value of this attribute denotes if base directory for STORE_FILE.
+     */
+    public static final String STORE_BASE = "store.base";
+
+    /**
+     * {@link KeystorePasswordStore} supported option
+     */
+    public static final String CREATE_STORAGE = "create.storage";
+    /**
+     * {@link KeystorePasswordStore} supported option
+     */
+    public static final String KEY_ALIAS = "key.alias";
+    /**
+     * {@link KeystorePasswordStore} supported option
+     */
+    public static final String KEY_PASSWORD = "key.password";
+    /**
+     * {@link KeystorePasswordStore} supported option
+     */
+    public static final String KEY_SIZE = "key.size";
+    /**
+     * {@link KeystorePasswordStore} supported option
+     */
+    public static final String CRYPTO_ALGORITHM = "crypto.algorithm";
+    /**
+     * {@link KeystorePasswordStore} supported option
+     */
+    public static final String RELOADABLE = "reloadable";
+    /**
+     * Default admin key alias for {@link KeystorePasswordStore}
+     */
+    public static final String DEFAULT_ADMIN_KEY_ALIAS = "ELY_ADMIN_KEY";
+
+    private static final String KEYSTORE_TYPE = "JCEKS";
+    private static final char[] CHARS = new char[0];
+    private static HashSet<String> supportedConfigurationAttributes = new HashSet<>();
+
+    static {
+        Collections.addAll(supportedConfigurationAttributes,
+                NAME, STORE_FILE, STORE_PASSWORD, STORE_BASE,
+                CREATE_STORAGE, KEY_ALIAS, KEY_PASSWORD,
+                KEY_SIZE, CRYPTO_ALGORITHM, RELOADABLE);
+    }
+
+    // used for reporting only / do not modify
+    private String storeName;
+    private boolean reloadable = false;
+    private boolean modifiable = false;
+    private File storeFile = null;
+    private String storeBase = "";
+    private char[] storagePassword = null;
+    private String adminKeyAlias = null;
+    private SecretKey adminKey = null;
+    private int keySize = 128;
+    private String cryptographicAlgorithm = "AES";
+    private boolean createStorage = false;
+
+    private KeyStore.ProtectionParameter adminKeyProtectionParam = null;
+
+    private final ConcurrentHashMap<String, Entry> storage = new ConcurrentHashMap<>();
+
+    /**
+     * Construct a new instance.
+     */
+    public KeystorePasswordStore() {
+    }
+
+    @Override
+    public void initialize(Map<String, String> attributes) throws CredentialStoreException {
+        storeName = attributes.getOrDefault(NAME, "myStore");
+        checkValidConfigurationAttributes(attributes.keySet());
+        storeBase = attributes.get(STORE_BASE);
+        storeFile = resolveFile(attributes.get(STORE_FILE), storeName);
+        storagePassword = convertPassword(attributes.get(STORE_PASSWORD));
```

Comment:
The STORAGE_PASSWORD option is for passing master password to the credential store to "unlock" it. Normally it it not used and master password is fetched through some kind of external gathering method. E.g. call script in the OS and use output as master password.